SetupCrossTenantRelationshipForTargetTenant.ps1

```diff
@@ -776,10 +776,58 @@ function UserPrompt() {
     $choice=$host.ui.PromptForChoice($title, $message, $options, 1)
     if ($choice -ne 0) {
         Exit}
-    else {Main}
+    Start-Sleep 2
+    Write-Host "`nWe are verifying that you are using the latest version of the script."`n
+    Write-Host "This requires that we download the latest version of the script from GitHub to compare with your local copy."
+    Write-Host "This file will be stored on your local computer temporarily, as well as overwrite your existing script file if it is out of date."
+    $title = "Confirm: Allow for download from GitHub and that you are running the script from the local directory the script exists in."
+    $message = "`nIf you are ready to begin this step, select 'Y'. `nIf you would prefer to manually download the scripts to make sure you have the latest version or change your path, select 'N'"
+    $yes = New-Object System.Management.Automation.Host.ChoiceDescription "&Yes", "Yes"
+    $no = New-Object System.Management.Automation.Host.ChoiceDescription "&No", "No"
+    $options = [System.Management.Automation.Host.ChoiceDescription[]]($yes, $no)
+    $choice=$host.ui.PromptForChoice($title, $message, $options, 1)
+    if ($choice -ne 0) {
+        Exit}
+    else {Verification}
+
 }
 
-UserPrompt
+function Verification {
+    Write-Host "`nBeginning verification steps."`n
+    Write-Host "Verifying ability to create a new organization relationship in the tenant."
+    try {
+        New-OrganizationRelationship -DomainNames contoso.onmicrosoft.com -Name Contoso -WhatIf -ErrorAction Stop
+    }
+    catch {
+        Write-Output "You need to run the command Enable-OrganizationCustomization before continuing with execution of the script."
+        Exit
+    }
+    Write-Host "`nVerifying that your script is up to date with the latest changes."
+    Write-Host "`nBeginning download of SetupCrossTenantRelationshipForTargetTenant.ps1 and creation of temporary folder."
+    if ((Test-Path -Path .\XTenantTemp) -eq $true) {
+        Remove-Item -Path .\XTenantTemp\ -Recurse -Force | Out-Null
+    }
+    New-Item -Path . -Name XTenantTemp -ItemType Directory | Out-Null
+    Invoke-WebRequest -Uri https://github.com/microsoft/cross-tenant/releases/download/Preview/SetupCrossTenantRelationshipForTargetTenant.ps1 -Outfile .\XTenantTemp\SetupCrossTenantRelationshipForTargetTenant.ps1
+    if ((Get-FileHash .\SetupCrossTenantRelationshipForTargetTenant.ps1).hash -eq (Get-FileHash .\XTenantTemp\SetupCrossTenantRelationshipForTargetTenant.ps1).hash) {

```

Comment:
nit: this also means versions != released versions are considered as non-latest which includes not published latest versions

Reply:
I'm not sure I follow? Can you elaborate.

Reply:
Do you mean like while I was testing I knew that I was always going to have a fail?